lib/policy_machine.rb

```diff
@@ -348,6 +380,20 @@ def bulk_persist
 
   private
 
+
+  def scoped_privileges_and_prohibitions(user_or_attribute, object_or_attribute, options)
+    if policy_machine_storage_adapter.respond_to?(:scoped_privileges)
+      policy_machine_storage_adapter.scoped_privileges(user_or_attribute.stored_pe, object_or_attribute.stored_pe, options).map do |op|
+        operation = PM::Operation.convert_stored_pe_to_pe(op, policy_machine_storage_adapter, PM::Operation)
+        [user_or_attribute, operation, object_or_attribute]
+      end
+    else
+      operations.grep(->operation{is_privilege_ignoring_prohibitions?(user_or_attribute, operation, object_or_attribute)}) do |op|
```

Comment:
So this selects the operations that satisfy the privilege triplet, ignoring prohibitions. I'm not sure passing a proc to Enumerable's `grep` option is the clearest way to do that.

```ruby
operations.each_with_object([]) do |op, memo|
  if is_privilege_ignoring_prohibitions?(user_or_attribute, op, object_or_attribute)
    memo << [user_or_attribute, op, object_or_attribute]
  end
end
```

Reply:
oh god trust me, I agree. This was original code I refactored from elsewhere, I don't like it but I don't think this PR is where I should change it